testing/test_postprocessing.py

```diff
@@ -64,49 +85,56 @@ def test_threshold(nii_seg):
     assert np.array_equal(nii_seg_proc.get_fdata()[4:7, 8, 4], np.array([0, 1, 1]))
 
 
-@pytest.mark.parametrize('nii_seg', [nii_dummy_seg()])
+@pytest.mark.parametrize('nii_seg', [nii_dummy_seg(), nii_dummy_seg(softseg=True)])
 def test_keep_largest_object(nii_seg):
     # Set a voxel to 1 at the corner to make sure it is set to 0 by the function
     coord = (1, 1, 1)
     nii_seg.dataobj[coord] = 1
     # Test function with array input
     arr_seg_proc = postproc.keep_largest_object(np.copy(np.asanyarray(nii_seg.dataobj)))
     assert isinstance(arr_seg_proc, np.ndarray)
+    assert check_bin_vs_soft(nii_seg.dataobj, arr_seg_proc)
     assert arr_seg_proc[coord] == 0
     # Make sure it works with nibabel input
     nii_seg_proc = postproc.keep_largest_object(nii_seg)
     assert isinstance(nii_seg_proc, nib.nifti1.Nifti1Image)
+    assert check_bin_vs_soft(nii_seg.dataobj, nii_seg_proc.dataobj)
     assert nii_seg_proc.dataobj[coord] == 0
 
 
-@pytest.mark.parametrize('nii_seg', [nii_dummy_seg()])
+@pytest.mark.parametrize('nii_seg', [nii_dummy_seg(), nii_dummy_seg(softseg=True)])
 def test_keep_largest_object_per_slice(nii_seg):
     # Set a voxel to 1 at the corner to make sure it is set to 0 by the function
     coord = (1, 1, 1)
     nii_seg.dataobj[coord] = 1
     # Test function with array input
     arr_seg_proc = postproc.keep_largest_object_per_slice(np.copy(np.asanyarray(nii_seg.dataobj)), axis=2)
     assert isinstance(arr_seg_proc, np.ndarray)
+    assert check_bin_vs_soft(nii_seg.dataobj, arr_seg_proc)
     assert arr_seg_proc[coord] == 0
     # Make sure it works with nibabel input
     nii_seg_proc = postproc.keep_largest_object_per_slice(nii_seg)
     assert isinstance(nii_seg_proc, nib.nifti1.Nifti1Image)
+    assert check_bin_vs_soft(nii_seg.dataobj, nii_seg_proc.dataobj)
     assert nii_seg_proc.dataobj[coord] == 0
 
 
-@pytest.mark.parametrize('nii_seg', [nii_dummy_seg()])
+@pytest.mark.parametrize('nii_seg', [nii_dummy_seg(), nii_dummy_seg(softseg=True)])
 def test_fill_holes(nii_seg):
     # Set a voxel to 0 in the middle of the segmentation to make sure it is set to 1 by the function
     coord = (7, 7, 4)
+    value_coord_copy = nii_seg.dataobj[coord]
```

Comment:
are you sure that this equality will not simply make `value_coord_copy` act as a pointer to the `nii_seg.dataobj` object, hence if the latter object is modified (which we want to assert), then `value_coord_copy` will be also modified , making the test below not sensitive to potential alteration of nibabel input object?